src/resources/breaking-changes.md

```diff
@@ -0,0 +1,821 @@
+
+### Language
+{: .no_toc}
+
+* **Language versioned**: Changed interpretation of [switch cases] from constant
+  expressions to patterns.
+
+* **Language versioned**: Class declarations from libraries that have been upgraded
+  to Dart 3.0 [can no longer be used as mixins by default][mixin class].
+
+* [Dart reports a compile-time error][50902] if a `continue` statement targets
+  a [label] that is not a loop (`for`, `do` and `while` statements) or a `switch`
+  member.
+
+### Libraries
+{: .no_toc}
+
+* The following existing classes have been made mixin classes:
+  `Iterable`, `IterableMixin`, `IterableBase`, `ListMixin`, `SetMixin`, `MapMixin`,
+  `LinkedListEntry`, `StringConversionSink`.
+
+#### `dart:core`
+
+* **Deprecated**: [Deprecated APIs][49529].
+
+#### `dart:async`
+
+* **Removed**: [Removed the deprecated][49529] [`DeferredLibrary`][] class.
+
+#### `dart:collection`
+
+* **Language versioned**: [Changes to platform libraries][collection].
+
+#### `dart:developer`
+
+* **Removed**: [Removed the deprecated][49529] [`MAX_USER_TAGS`][] constant.
+  Use [`maxUserTags`][] instead.
+* **Removed**: [Removed the deprecated][50231] [`Metrics`][], [`Metric`][], [`Counter`][],
+  and [`Gauge`][] classes as they have been broken since Dart 2.0.
+
+#### `dart:html`
+
+* **Removed**: [Removed the deprecated `registerElement` and `registerElement2`][49536]
+  methods in `Document` and `HtmlDocument`.
+
+#### `dart:math`
+
+* **Language versioned**: The `Random` interface can only be implemented,
+  not extended.
+
+#### `dart:io`
+
+* [Updated `NetworkProfiling`][51035] to accommodate new `String` ids that are
+  introduced in vm_service:11.0.0
+
+[dart3]: /resources/dart-3-migration/
+[switch cases]: https://dart.dev/language/branches#switch
+[mixin class]: https://dart.dev/language/mixins#class-mixin-or-mixin-class
+[label]: https://dart.dev/language/branches#switch
+[50902]: https://github.com/dart-lang/sdk/issues/50902
+[collection]: /resources/dart-3-migration#dartcollection
+[49529]: https://github.com/dart-lang/sdk/issues/49529
+[`DeferredLibrary`]: https://api.dart.dev/stable/2.18.4/dart-async/DeferredLibrary-class.html
+[`deferred as`]: https://dart.dev/guides/language/language-tour#deferred-loading
+[`MAX_USER_TAGS`]: https://api.dart.dev/stable/dart-developer/UserTag/MAX_USER_TAGS-constant.html
+[`maxUserTags`]: https://api.dart.dev/beta/2.19.0-255.2.beta/dart-developer/UserTag/maxUserTags-constant.html
+[50231]: https://github.com/dart-lang/sdk/issues/50231
+[`Metrics`]: https://api.dart.dev/stable/2.18.2/dart-developer/Metrics-class.html
+[`Metric`]: https://api.dart.dev/stable/2.18.2/dart-developer/Metric-class.html
+[`Counter`]: https://api.dart.dev/stable/2.18.2/dart-developer/Counter-class.html
+[`Gauge`]: https://api.dart.dev/stable/2.18.2/dart-developer/Gauge-class.html
+[49536]: https://github.com/dart-lang/sdk/issues/49536
+[51035]: https://github.com/dart-lang/sdk/issues/51035
+
+
+## Released in 2.19.0
```

Comment:
This isn't necessary, but a question/consideration:

I wonder if it's worth it to explicitly set the IDs of the version headers for easier to remember/write link targets. For example, this one could be:

```
## Released in 2.19.0 {#2-19-0}
```

Then the link would just be https://dart.dev/resources/breaking-changes#2-19-0

Reply:
Great idea! Will do

Reply:
Are we sure that's how that works? 😅 
![image](https://github.com/dart-lang/site-www/assets/111139605/cb553943-9225-434c-80fa-658476aaa45c)

Reply:
Maybe that's what I meant 👀 ...

Emm after some testing, it seems overriding the ID doesn't work if the override begins with a number. I think it's fine to drop this idea for now.

Reply:
I figured it was the numbers. Oh well, good idea while it lasted!